scripts/subsystem_versions.py

```diff
@@ -0,0 +1,69 @@
+"""This script is used to get the versions of the submodules."""
+
+import re
+import sys
+import json
+import argparse
+import subprocess
+
+VERSION_REGEX = re.compile("v([0-9])")
+SUBSYSTEMS = [
+    "head",
+    "gantry-x",
+    "gantry-y",
+    "gripper",
+    "pipettes-single",
+    "pipettes-multi",
+    "pipettes-96",
+    "pippetes-384",
+]
+
+
+def main(args):
+    version_info = {}
+    for subsystem in SUBSYSTEMS:
+        if args.target == "all" or subsystem in args.target:
+            try:
+                version = (
```

Comment:
i get that this is more extensible but we really are building everything off the same commit, so let's hoist this logic out of the loop to make it faster?